application.py

```diff
@@ -295,31 +296,47 @@ def all_movie_ratings():
+@application.route('/rating', methods = ['POST'])
+def movie_option():
+    userID = request.json['id']
+
     #this endpoint will take the user id and look through all the movies in the database that do not have a Yes/No choice made already by the user
     #it will then output one of those undecided options for the movie
 
     #first query to get the user that is currently logged in
     m = User.query.filter_by(userID=userID).first()
     
-    #this needs to be expanded in future sprints
-
-    #the results scanner is hardcoded for Sprint 1 to take in the first instace in the User_Rates_Movies index to see if the user has any movies they have alreadty watched
-    results = [m.moviesRated[1].movieID, m.moviesRated[1].userID, m.moviesRated[1].isLiked]
-
-    #next, we get movies that are not on the list to get the movies that have not been rated
-    moviesUnrated = Movie.query.filter(Movie.movieID != results[0]).first()
-    movieData = [moviesUnrated.name, moviesUnrated.genre]
-
-    #finally, return the data of the movie being rated, the userID, and the movie info to display
-    fullResults = [results[0],results[1],movieData[0],movieData[1]]
-
-    #next 2 lines are for unit testing of the arrays that we will send to the JS
-    #print (results)
-    #print (movieData)
-   
-    return jsonify(fullResults)
+    #logic for finding a movie
+    #1. Get a list of all rated movies for a user and we convert this into a list of movieIds only
+    
+    #declare list variable to add movieIDs to
+    ratedMovies = []
+    
+    #2. For each movie-user-rated relationship, we append the movieID to out list of ratedMovies
+    for i in m.moviesRated:
+        ratedMovies.append(i.movieID)
+
+    #3. Get a list of all the movies
+    allMovieID = []
+    allMovies = Movie.query.all()
+    for j in allMovies:
+        allMovieID.append(j.movieID)
+
+    #4. Find the movies that are in the list of all movies but not in the list of the movies rated by the user (this can be done by comparing the 2 lists)
+    unratedMovies = np.setdiff1d(allMovieID,ratedMovies)
+
+    #5. Gather the relevant details of the first movie on that list from 4 to output
+    #right now we use a simple selection of the first movie on the list.  Later on, if we feel that more complex logic is needed to create a fair representation, then that can be changed below
+    #add in a try/except in case all movies have been rated
+    try:
+        nextMovie = Movie.query.get(unratedMovies[0])
+        #clean this up so that we are displaying in JSON format instead of arrays bc that is more intuitive
+        movieData = {'userID':userID, 'movieID':nextMovie.movieID, 'movie_name':nextMovie.name, 'movie_genre':nextMovie.genre, 'movie_director':nextMovie.director}
+    except:
+        movieData = {'userID':userID, 'movieID':'N/A', 'movie_name':'N/A', 'movie_genre':'N/A', 'movie_director':'N/A'}
+    
+    #finally, we return the results
+    return jsonify(movieData)
```

Comment:
@Great commenting in this endpoint. Nice work Rafiq!  💯

Reply:
Thanks!